app/controllers/events_controller.rb

```diff
@@ -0,0 +1,97 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  before_action :user_login
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+    @users = User.all
+    @today = Event.today
+    @upcoming = Event.upcoming
+    @previous = Event.previous
+  end
+
+  # GET /events/1
+  # GET /events/1.json
+  def show
+    set_event
+    @users = User.all
+    @user = User.find(@event.creator.id)
+    @attendees = Attendance.where(event_id: @event.id)
+  end
+
+  def attend
+    set_event
+    @attendee = Attendance.new
+    @attendee.event_id = @event.id
+    @attendee.user_id = current_user.id
+    @attendee.save
+
+    redirect_to @event
+  end
+
+  # GET /events/new
+  def new
+    @event = current_user.events.build
+  end
+
+  # GET /events/1/edit
+  def edit; end
+
+  # POST /events
+  # POST /events.json
+  def create
+    @event = current_user.events.build(event_params)
+    @event.creator_id = current_user.id
```

Comment:
:bulb: Well, we are using the build method here. Thus, we don't need the line 46 `@event.creator_id = current_user.id`. It should automatically assign the current users id at line 45. Check and see :blush: